### PR TITLE
fix: Fade out navigation when editing and mouse hasn't moved

### PR DESCRIPTION
**Pull request by iamsaumya**
towards #2976

### CODE REVIEW

File: app/hooks/useMouseMove.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import * as React from "react";
+
+const useMouseMove = () => {
```
Comment (tommoor) [inline app/hooks/useMouseMove.ts:3]:
```suggestion
const useMouseMove = (timeout: number = 5000) => {
```

Lets make the `5000` an argument

Comment (tommoor) [inline app/hooks/useMouseMove.ts:3]: Quick documentation comment? Trying to do this with all new hooks/methods

File: app/scenes/Document/components/Header.tsx [hunk 2]
```diff
@@ -337,4 +343,10 @@ const Status = styled(Action)`
   color: ${(props) => props.theme.slate};
 `;
 
+const Wrapper = styled(Flex)<{ $hideHeader: boolean }>`
+  opacity: ${(props) => (props.$hideHeader ? 0 : 1)};
+  visibility: ${(props) => (props.$hideHeader ? "hidden" : "visible")};
```
Comment (tommoor) [inline app/scenes/Document/components/Header.tsx:348]: Intrigued as to why `visibility` was also needed? I'm sure there's a good reason

File: app/scenes/Document/components/Header.tsx [hunk 3]
```diff
@@ -211,7 +217,7 @@ function DocumentHeader({
           </>
         }
         actions={
-          <>
+          <Wrapper $hideHeader={hideHeader}>
             <ObservingBanner />
```
Comment (tommoor) [inline app/scenes/Document/components/Header.tsx:221]: Lets keep `<ObservingBanner />` visible

**Review by tommoor — COMMENTED** [review]
Nice, I think we'll also end up making this potentially a user preference but we'll see what feedback comes in

File: app/scenes/Document/components/Header.tsx (hunk 2 above)
Comment (iamsaumya) [reply app/scenes/Document/components/Header.tsx:348]: To not trigger the click/hover events.

File: app/scenes/Document/components/Header.tsx (hunk 3 above)
Comment (iamsaumya) [reply app/scenes/Document/components/Header.tsx:221]: Cool. Btw the navigation is always visible in collab editing mode. Do we want to fade out in it too?

Comment (tommoor) [reply app/scenes/Document/components/Header.tsx:221]: Yep, same behavior in collab is ideal

File: app/scenes/Document/components/Header.tsx [hunk 4]
```diff
@@ -163,6 +170,35 @@ function DocumentHeader({
...
+        placement="bottom"
+      >
+        <Button icon={<PlusIcon />} {...props} neutral>
+          {t("New doc")}
+        </Button>
+      </Tooltip>
+    ),
+    [t]
+  );
+
```
Comment (iamsaumya) [inline app/scenes/Document/components/Header.tsx:201]: Inline usage to rerendering in collab mode.

File: app/editor/components/EmojiMenu.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ const searcher = new FuzzySearch<{
   sort: true,
 });
 
-class EmojiMenu extends React.Component<
+class EmojiMenu extends React.PureComponent<
```
Comment (iamsaumya) [inline app/editor/components/EmojiMenu.tsx:24]: It takes 10ms to render even when it doesn't render any of the children's 😮
Screenshot 2022-05-01 at 4 27 32 AM

**Comment by tommoor** [comment]
After playing with it, it feels quite good – a couple of tweaks:

- The fade out could be a touch slower, maybe another 500ms.
- I think we should also fade out the "References" area at the bottom of the doc that shows nested and backlinked docs
- I'm wondering if this is contentious enough to make it a user preference, thoughts?

**Comment by iamsaumya** [comment]
> I'm wondering if this is contentious enough to make it a user preference, thoughts?

Sounds good. I think we will have to create a preferences page. Do you have a page in mind where we can put this?

**Comment by tommoor** [comment]
I can add user preferences in the next week, just need to figure out how I want to store them 😋

**Comment by iamsaumya** [comment]
Could we review and get this merge? Next week, we can always add this to the preference page - I assume we would keep it enabled by default.

**Review by tommoor — APPROVED** [review]

**Pull request merged**